### PR TITLE
CLIENT-6539 | Getting audio levels and add insights for extensions

### PR DESCRIPTION
**Pull request by charliesantos**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### JIRA link(s):

- [CLIENT-6539](https://issues.corp.twilio.com/browse/CLIENT-6539)

### Description

- Get audio levels without relying on requestAnimationFrame which won't trigger when a tab is inactive or when running as an extension.
- Adding insights whether running as an extension.

### Before review
* [x] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with `npm run build:release`
* [x] Manually sanity tested running locally
* [x] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Squashed erroneous commits if necessary
* [ ] Re-tested if necessary

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+1.8.1 (In Progress)
+====================
+
+Bug Fixes
+---------
+
+* Fixed an issue where audio levels reported are zero when running as an extension, or when the browser tab is inactive or minimized.
```
Comment (ryan-rowland) [inline CHANGELOG.md]: "Fixed an issue causing audio levels to be reported as zero when running as an extension, or when the browser tab is inactive or minimized"

File: lib/twilio/connection.ts [hunk 2]
```diff
@@ -1380,10 +1380,8 @@ namespace Connection {
   declare function rejectEvent(connection: Connection): void;
 
   /**
-   * Emitted on `requestAnimationFrame` (up to 60fps, depending on browser) with
-   *   the current input and output volumes, as a percentage of maximum
-   *   volume, between -100dB and -30dB. Represented by a floating point
-   *   number.
+   * Emitted up to 20hz with the current input and output volumes, as a percentage of maximum
```
Comment (ryan-rowland) [inline lib/twilio/connection.ts]: I don't think `hz` is common outside of monitor refresh rates, is it? How about `Emitted every 50ms`?

File: lib/twilio/device.ts [hunk 3]
```diff
@@ -277,6 +277,11 @@ class Device extends EventEmitter {
     [Device.SoundName.Outgoing]: true,
   };
 
+  /**
+   * Whether SDK is run as a browser extension
+   */
+  private _isBrowserExtension: boolean;
```
Comment (ryan-rowland) [inline lib/twilio/device.ts]: Does this catch extensions for all browsers or just Chrome?

File: lib/twilio/device.ts [hunk 4]
```diff
@@ -797,6 +808,7 @@ class Device extends EventEmitter {
    */
   private _createDefaultPayload = (connection?: Connection): Record<string, any> => {
     const payload: Record<string, any> = {
+      browser_extension: this._isBrowserExtension,
```
Comment (ryan-rowland) [inline lib/twilio/device.ts]: Prefer `is_extension`. We will need to add this to the Insights schema as well.

File: lib/twilio/rtc/peerconnection.js [hunk 5]
```diff
@@ -8,6 +8,8 @@ const Log = require('../log');
 const util = require('../util');
 const RTCPC = require('./rtcpc');
 
+// 20hz
```
Comment (ryan-rowland) [inline lib/twilio/rtc/peerconnection.js]: Nit: Let's remove this comment.

File: lib/twilio/rtc/peerconnection.js [hunk 6]
```diff
@@ -8,6 +8,8 @@ const Log = require('../log');
...
+const EMIT_VOLUME_TIMEOUT = 50;
```
Comment (ryan-rowland) [inline lib/twilio/rtc/peerconnection.js]: Prefer `VOLUME_INTERVAL_MS`

File: lib/twilio/connection.ts (hunk 2 above)
Comment (charliesantos) [reply lib/twilio/connection.ts]: I was thinking in terms of frequency. But yes, `in ms` would be clearer.

File: lib/twilio/device.ts (hunk 3 above)
Comment (charliesantos) [reply lib/twilio/device.ts]: It will catch chrome, edge, and firefox. Safari doesn't support the same extension model.

**Pull request merged**